### ISSUE TITLE
 Refactor `SOLD2Net` to Support Dataclasses for Configuration

### ISSUE THREAD

**Issue by lappemic** [kornia/kornia#2908]
In the process of updating dictionary configurations to dataclasses (see #2093), I noticed that `kornia.feature.sold2.backbones.SOLD2Net` currently also only supports dictionaries. This restriction was addressed by using the `dataclass_to_dicts` function a previous commit ([see details](https://github.com/kornia/kornia/pull/2880/commits/53673fde0465bdd76c4d189d33ae64aa1e2f7b3c)). To maintain consistency with the ongoing dataclasses for config refactoring effort, should `SOLD2Net` be updated to accept dataclasses directly?

**Comment by johnnv1** [comment]
Yes, although it doesn't matter as much because it's more for “internal” use, I think it's great to keep it consistent -- and I think it's more legible
